lgc/patch/PatchLoadScalarizer.cpp

```diff
@@ -182,7 +182,13 @@ void PatchLoadScalarizer::visitLoadInst(LoadInst &loadInst) {
       Value *loadCompPtr = m_builder->CreateConstGEP1_32(compTy, newLoadPtr, i,
                                                          loadInst.getPointerOperand()->getName() + ".i" + Twine(i));
       // Calculate the alignment of component i
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 426675
+      // Old version of the code
       uint64_t compAlignment = MinAlign(loadInst.getAlignment(), i * compSize);
+#else
+      // New version of the code (also handles unknown version, which we treat as latest)
+      uint64_t compAlignment = MinAlign(loadInst.getAlign().value(), i * compSize);
+#endif
```

Comment:
Calling `.value()` on an alignment is discouraged, it’s better to keep the values as `Align`. You can do
```cpp
Align compAlignment = commonAlignment(loadInst.getAlign(), Align(i * compSize));
```

instead. Also, I think the if guards are not needed.

Reply:
Hmm, I get an assertion with those changes - obviously something else is required. I'll dig further.